### DISCUSSION TITLE
Mobile device conversation error

### DISCUSSION THREAD

**Discussion by krinlove99** [open-webui/open-webui#3372] [General]
When I used a mobile device to use my openwebui and conduct a conversation, the following error was prompted in the reply: Invalid regular expression: invalid group specifier name (the reply from AI can usually return a few words and truncate normally)

I have confirmed that this problem will occur when using all channel source models through the mobile terminal.
I have confirmed that this problem will not occur on the PC side.
There is no error in the console, and all return 200.

Deployment device: huggingface (from the official template) and ubuntu 20 
Used device: iOS iPadOS

**Comment by wanderingmeow** [comment]
This error was first observed in version 0.3.5, but the underlying regular expression was added as early as version 0.1.x.

https://github.com/open-webui/open-webui/blob/9e4dd4b86f77653b59dc1fa97fe3f72e8252b359/src/lib/utils/index.ts#L462

As mentioned in this Stack Overflow question: https://stackoverflow.com/questions/51568821/works-in-chrome-but-breaks-in-safari-invalid-regular-expression-invalid-group, regex lookbehind doesn't work on Safari before version 16.4 (iOS 16.4), which was released in Spring 2023.

To resolve this issue, changing the line to the following one should fix the problem.
```javascript
const sentences = text.match(/[^.?!]+[.!?]+[\])'"`’”]*|.+/g) || [];
```

This is not only compatible with older version of Safari but also way faster in Safari than the lookbehind implementation. I have benchmarked on a long text:

|Browser|Match vs Lookbehind|
|:-|-:|
|Safari 17.5|16.2x|
|Chromium 125|2.2x|
|Firefox 127|1.4x|

Reference: https://stackoverflow.com/questions/11761563/javascript-regexp-for-splitting-text-into-sentences-and-keeping-the-delimiter

**Reply by krinlove99** [reply]
Thanks for your answers🤣，it works

**Reply by justinh-rahb** [reply]
Not sure how we missed this one, there was a prior PR for it. We'll get this fixed as soon as able.

**Reply by kylespencer** [reply]
Great. Can someone please explain how to apply this fix to Docker-based deployments?

**Reply by CloneXpert** [reply]
I would be curious to know also. I can't match the code referenced here anymore, when looking at the current main. What would be the easiest way  to patch this locally, or whatever makes sense?

**Reply by wanderingmeow** [reply]
```bash
find ./build/_app/immutable/chunks/index.*.js -exec sed -i -e "s/.split(new RegExp(\"(?<=\[.\!?\])\\\\\\\\s+\"));/.match(\\/\\[^.?\!\]+\[.\!?\]+\[\\\\])'\"\`\’\”\]\*|.+\/g)||\[\];/g" {} \;
```

**Reply by CloneXpert** [reply]
Very impressive, I didn't expect that. Thanks so much for giving magic bullet.

**Comment by cory2005** [comment]
I'm still having the same issue, the docker image was updated yesterday

**Reply by cory2005** [reply]
I applied the fix as mentioned in the post above "find ./build.." however it still wouldn't work in chrome on ios, however it did work after applying the fix when I tried it in safari

**Comment by kylespencer** [comment]
Running into this issue with iPhone users on Safari & Chrome & Firefox.

**Comment by krinlove99** [comment]
https://github.com/open-webui/open-webui/discussions/3372#discussioncomment-9889979

You can refer to this comment and need to modify the mentioned code because a certain version of the library for iOS.
I have confirmed that this solution works.
Reference: https://stackoverflow.com/questions/11761563/javascript-regexp-for-splitting-text-into-sentences-and-keeping-the-delimiter

**Comment by tijmen** [comment]
Could this get merged in, please? Currently as of v0.3.31 (1 hour ago) open-webui is not usable for ***old*** iPhone users.

**Comment by daxiass** [comment]
I'm also having this issue, but it hasn't been updated until today...
Changing docker is such a hassle
I'm so sad

**Reply by AgatheBauer** [reply]
Updating to iOS 18 worked for me. 16.4 or newer seems to work too.
